x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/GetEngineAction.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class GetEngineAction extends ActionType<GetEngineAction.Response> {
+
+    public static final GetEngineAction INSTANCE = new GetEngineAction();
+    public static final String NAME = "cluster:admin/xpack/engines/get";
```

Comment:
I don't think this is the correct privilege, but I could use some guidance here. I started with `indices:data/read/engines/get` however this leads to errors since the only allowed indices calls that don't specify the indices are scroll and async. Since we don't want to change core internals if we can help it, what would be the best privilege to add here?

Reply:
Since we create the alias at Engine creation we can use `indices:admin/engine/get"`. For indices action, the alias/indices in the request are intercepted by the security layer to check that the user has the permission on them. 
To enable this behaviour, the `Request` below should extend `IndicesRequest.Replaceable` and implements `String[] indices()` and `indices(String... indices)`. Since Engine is not an index, we should return the alias name in `indices()` to handle the security check. 
Does that make sense?

Reply:
Yes, that does! Thank you for the pointer and explanation!